api/src/main/java/org/openmrs/api/impl/ProviderServiceImpl.java

```diff
@@ -302,4 +303,48 @@ public Provider getUnknownProvider() {
 		return getProviderByUuid(Context.getAdministrationService().getGlobalProperty(
 		    OpenmrsConstants.GP_UNKNOWN_PROVIDER_UUID));
 	}
+
+	@Override
+	@Transactional(readOnly = true)
+	public List<ProviderRole> getAllProviderRoles(boolean includeRetired) {
+		return dao.getAllProviderRoles(includeRetired);
+	}
+
+	@Override
+	@Transactional(readOnly = true)
+	public ProviderRole getProviderRole(Integer id) {
+		return dao.getProviderRole(id);
+	}
+
+	@Override
+	@Transactional(readOnly = true)
+	public ProviderRole getProviderRoleByUuid(String uuid) {
+		return dao.getProviderRoleByUuid(uuid);
+	}
+
+	@Override
+	@Transactional
+	public ProviderRole saveProviderRole(ProviderRole role) {
+		return dao.saveProviderRole(role);
+	}
+
+	@Override
+	@Transactional
+	public void retireProviderRole(ProviderRole role, String reason) {
+		// BaseRetireHandler handles retiring the object
```

Comment:
Are we going to be putting these comments in every service method where we retire or unretire objects?